scaladoc/src/dotty/tools/scaladoc/site/customTags/IncludeTag.scala

```diff
@@ -0,0 +1,88 @@
+package dotty.tools.scaladoc.site.tags
+
+
+import liqp.TemplateContext
+import liqp.nodes.LNode
+import liqp.tags.Tag
+import scala.io.Source
+import java.io.File
+import scala.util.{Try, Success, Failure}
+import scala.jdk.CollectionConverters._
+
+class IncludeTag extends Tag("include") {
+
+  override def render(context: TemplateContext, args: Array[? <: LNode]): Object = {
+    // Render arguments and validate filename
+    val filename = args(0).render(context) match {
+      case s: String if s.endsWith(".html") => s
+      case s: String => throw new IllegalArgumentException(s"{% include %} tag requires a string argument ending with .html, but got: $s")
+      case _ => throw new IllegalArgumentException("{% include %} tag requires a string argument ending with .html")
+    }
+
+    val filePath = s"${IncludeTag.docsFolder}/$filename"
+    println(s"Attempting to include file: $filePath")
+
+    val inputData = args(1).render(context)
+    context.put("include", inputData)
+
+    try {
+      // Read and parse the file content
+      val fileContent = readFileContent(filePath) match {
+        case Success(content) => content
+        case Failure(exception) => throw new IllegalArgumentException(s"Error reading file '$filePath': ${exception.getMessage}")
+      }
+
+      // Resolve nested includes within the file content
+      val resolvedContent = resolveNestedIncludes(fileContent, context)
+
+      val siteData = context.getVariables().asInstanceOf[java.util.Map[String, Any]]
+      // Parse the resolved content
+      context.getParser().parse(resolvedContent).render(siteData)
+    } finally {
+      // Clean up: remove the input data from the context
+      context.remove("include")
+    }
+  }
+
+  private def readFileContent(filePath: String): Try[String] = {
```

Comment:
minor: it seems that every use case of this function always rethrows this error. Can we just not catch it in the first place?